cypress/integration/dunning/create_dunning_candidates.js

```diff
@@ -0,0 +1,211 @@
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+/// <reference types="Cypress" />
+
+
+import {SalesInvoice, SalesInvoiceLine} from "../../support/utils/sales_invoice";
+import {getLanguageSpecific} from "../../support/utils/utils";
+import {DocumentActionKey, DocumentStatusKey} from "../../support/utils/constants";
+import {BPartner} from "../../support/utils/bpartner";
+import {DunningCandidates} from "../../page_objects/dunning_candidates";
+import {applyFilters, selectNotFrequentFilterWidget, toggleNotFrequentFilters} from "../../support/functions";
+import {DunningType} from "../../support/utils/dunning_type";
+
+describe('Create Dunning Candidates', function () {
+  // human readable date with millis!
+  const date = new Date(Date.now() - new Date().getTimezoneOffset() * 60000).toISOString();
+
+  const dunningTypeName = `Dunning ${date}`;
+  // const dunningTypeName = `Dunning 2019-07-05T10:09:30.514Z`;
+
+
+  const businessPartnerName = `Customer Dunning ${date}`;
+  const paymentTerm = 'immediately';
+
+  const salesInvoiceTargetDocumentType = 'Sales Invoice';
+  const productName = 'Convenience Salat 250g';
+  const originalQuantity = 200;
+
+
+  // Test data
+  let siDocumentNumber;
+  let siCurrency;
+  let siDueDate;
+  let siTotalAmount;
+
+
+  before(function () {
+    // This wait is stupid.
+    // It also appears to be a good workaround for the problems in
+    // cypress/support/utils/utils.js:1
+    cy.wait(5000);
+  });
+
+  it('Prepare dunning type', function () {
+
+    cy.fixture('settings/dunning_type.json').then(dunningType => {
+      Object.assign(new DunningType(), dunningType)
+        .setName(dunningTypeName)
+        .apply();
+    });
+  });
+
+  it('Prepare customer bpartner (via api)', function () {
+    cy.fixture('sales/simple_customer.json').then(customerJson => {
+      const bpartner = new BPartner({...customerJson, name: businessPartnerName})
+        .setCustomer(true)
+        .setDunning(dunningTypeName)
+        .setPaymentTerm(paymentTerm)
+        .setBank(undefined);
+
+      bpartner.apply();
+    });
+  });
+
+  it('Prepare sales invoice', function () {
+    cy.fixture('sales/sales_invoice.json').then((salesInvoiceJson) => {
+      new SalesInvoice(businessPartnerName, salesInvoiceTargetDocumentType)
+        .addLine(
+          new SalesInvoiceLine().setProduct(productName).setQuantity(originalQuantity)
+          // todo @dh: how to add a "per test" packing item
```

Comment:
i think these commented out lines are not useful in the spec; 
I.e. i don't think we plan to test dunning particularly on packing materials in the future

Reply:
fixed